sale_order_amount_history/models/sale_order_amount_history.py

```diff
@@ -59,3 +62,11 @@ def _compute_amount_company(self):
             rec.amount_diff_company = rec.currency_id._convert(
                 rec.amount_diff, rec.company_currency_id, rec.company_id, rate_date
             )
+
+    @api.depends("partner_id")
```

Comment:
```suggestion
    @api.depends(
        "partner_id",
        "partner_id.commercial_partner_id",
        "partner_id.industry_id",
        "partner_id.commercial_partner_id.industry_id",
    )
```

To be able to reflect the changes of the partner structure/industry assignments as they happen.

The down side of this:
* There can possibly be a large volume of computation when the master records are changed
* Customer might actually have a different idea (e.g. once a record is created for reporting, the content should not be changed).

What do you think?